doc/CI_Hardware.md

```diff
@@ -0,0 +1,286 @@
+            access-points:
+                "SSID_name":
+                    password: "WiFi_password"
+```
+2. Generate and apply new network plan
+```sudo netplan generate; sudo netplan apply```
+
+### Action runner: **command not found**
+
+If you experience issue with the command not being found, but you are very sure that you have installed the corresponding software. Try to uninstall the runner and reinstall it.
+
+```bash
+# 1. Uninstall Action Runner Server
+$ cd ~/actions-runner
+$ sudo ./svc.sh stop
+$ sudo ./svc.sh uninstall
+$ ./config.sh remove
+
+# 2. Reinstall Action Runner Server
+# Follow the guide in Getting Started.
+```
+**Note**: Possible reason to this issue is that the server environment and path varaible cannot be update after it has started. Thus, it requires a reconfiguration to included the updated path and environment variables.
+
+## Creating the Configuration File
+After all is set up, the first step in creating an instance is to run the [runner_init](https://github.com/goodoomoodoo/tock-test-harness/blob/main/runner_init.py) script to create a configuration file for the raspberry pi you just set up.
+This configuration file will then be used in the workflow of actions for github whenever an update is pushed to be accessed by the runners, which will be denoted as config.toml for simplicity sake here.
+
+What this **configuration file** contains is:
+- Title
+    - Name of the configuration. 
+    - This can be anything you want but make sure to title it appropriately.
+- Board name
+    - This will be provided when the executable is called on what boards are supported. Choose which board you are using that is supported. You can NOT use a board that is not supported by the Tock OS. 
+- Board Path
+    - This is the path of the Tock supported board in the official tock repository. The path must be provided when setting up the configuration. This is essentially the directory relative to the Tock boards directory. 
+- Harness ID
+    - This is the specific Identity Number of the Raspberry Pi. This will be used in the test.config.toml which is kept for each board type to run tests on in the runner. 
+- Communication Protocol 
+    - Used to run tockloader
+    - The list of communication protocols are provided, and one chosen must be in the list. 
+        - Typically Jlink is the one chosen here
+
+When runner_init script is executed, there will be prompts to fill in this information and the configuration file will be created based on the responses, filling in the file with the information listed above.
+
+**Example Configuration File:**
+
+```toml
+title = "pi_test"
+
+[env]
+board = "nrf52dk"
+path = "{home}/actions-runner/_work/tock/tock/boards/nordic/nrf52dk/"
+harness_id = "{Pi ID}"
+communication_protocol = "jlink"
+```
+
+**Note:** {home} and {Pi ID} would depend on your configuration, where {home} will be your home directory to start the path, and {Pi ID} will be your specific Raspberry Pi ID
```

Comment:
Where is Pi ID assigned? Is that something set on the RPi or is it something set in GitHub somewhere?